src/dvclive/live.py

```diff
@@ -265,11 +265,18 @@ def _init_dvc(self):  # noqa: C901
             self._include_untracked.append(self.dir)
 
     def _init_dvc_file(self) -> str:
-        if isinstance(self._dvcyaml, str):
-            if os.path.basename(self._dvcyaml) == "dvc.yaml":
-                return self._dvcyaml
-            raise InvalidDvcyamlError
-        return "dvc.yaml"
+        if self._dvcyaml is None:
+            return "dvc.yaml"
+        if isinstance(self._dvcyaml, bool):
```

Comment:
hmm, so is it something new? do we want to support bool as well?

we need to update docs then?

what happens if it's False (sounds like `dvc.yaml` should not be produced at all then?)

Reply:
On True/False/None, it defaults to "dvc.yaml".
I had to make this change to pass an already written test case, which tests`True`, `False` values to  `dvcyaml` argument.

Reply:
@AbdullahMakhdoom could you please point me to that test?

Reply:
@shcheklein, you can read the docstring for `dvcyaml`.

Reply:
okay, so:

```
                If `False`, DVCLive will not write to "dvc.yaml" (useful if you are
                tracking DVCLive metrics, plots, and parameters independently and
                want to avoid duplication).
```

it means we can't default to `dvc.yaml` - we should probably not running this function at all then if it's False.

Reply:
@shcheklein The code is maybe a bit of a mess here, but this sets `self._dvcfile`, which shouldn't matter for whether the `dvc.yaml` file is written. The code still checks `self._dvcyaml` to decide whether to write `dvc.yaml` [here](https://github.com/iterative/dvclive/blob/68884628e5845f95b89c950016d1b9a1a190cb60/src/dvclive/live.py#L427-L428). This is tested here:

https://github.com/iterative/dvclive/blob/68884628e5845f95b89c950016d1b9a1a190cb60/tests/test_make_dvcyaml.py#L461-L463

It shouldn't matter much if `self._dvcfile` is set in this scenario, but the slight justification for still setting it to `dvc.yaml` is that if someone has set `Live(dvcyaml=False)` but then manually calls `make_dvcyaml()`, it will still write to `dvc.yaml`.

Reply:
It's unclear why using `False` is necessary here, as setting `dvcyaml` to `None` appears to produce the same outcome. Both tests ensure that the `dvc.yaml` file is not created when `dvcyaml` is either False or None, indicating that these options are functionally equivalent in this context. 

```

def test_make_dvcyaml_false(tmp_dir, mocker):
    dvclive = Live("logs", dvcyaml=False)
    dvclive.end()

    assert not os.path.exists("dvc.yaml")


def test_make_dvcyaml_none(tmp_dir, mocker):
    dvclive = Live("logs", dvcyaml=None)
    dvclive.end()

    assert not os.path.exists("dvc.yaml")

Reply:
Also, apologize for the delayed responses, I've been quite busy with my day job. Please let me know the best way to move forward, and I'll make sure to do my best.

Reply:
No problem @AbdullahMakhdoom, thanks for keeping on top of this!

`dvcyaml` used to be a bool argument and support for bools was left to avoid making a breaking change. We should probably have dropped this during the last major release but missed it.